project/apps/web/lib/api/auth.ts

```diff
@@ -2,7 +2,15 @@ import { create } from "zustand";
 import { apiCall } from "./apiClient";
 import { SignInDto, SignUpDto } from "@repo/dtos/auth";
 import { persist } from "zustand/middleware";
-import { UserDetails } from '../../../api-gateway/src/supabase/collection';
+
+// TODO: fix this interface by prob a DTO or sth
+export interface UserDetails {
+  created_at: string;
+  email: string;
+  id: string;
+  role: string;
+  username: string;
+}
```

Comment:
Added to issue #27, will look into this once merged

Reply:
actually can do this @jeffrey-jian @gongg21 
```
import { Tables } from '@repo/dtos/generated/types/auth.types';
export type UserDetails = Tables<'profiles'>;
```
See [here](https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g30/blob/main/project/apps/user-service/src/supabase/collection.ts)